src/shared/sidepanel/sidepanel-apis.ts

```diff
@@ -0,0 +1,52 @@
+import browser from 'webextension-polyfill';
+import { getSidepanelUrl } from '../getPopupUrl';
+import { invariant } from '../invariant';
+
+export async function getActiveTab() {
+  const tabs = await browser.tabs.query({ active: true, currentWindow: true });
+  return tabs.length ? tabs[0] : null;
+}
+
+async function resolveOpenOptions(
+  openOptions?: chrome.sidePanel.OpenOptions
+): Promise<chrome.sidePanel.OpenOptions> {
+  if (openOptions) {
+    return openOptions;
+  } else {
+    const tab = await getActiveTab();
+    invariant(tab, 'Active tab not found');
+    invariant(tab.id, 'Active tab id not set');
+    return { tabId: tab.id };
+  }
+}
+
+export async function openSidePanel({
+  pathname,
+  searchParams,
+  openOptions,
+}: {
+  pathname: string;
+  searchParams: URLSearchParams | null;
+  openOptions?: chrome.sidePanel.OpenOptions;
+}) {
+  const url = getSidepanelUrl();
+  // TODO: support templateType=sidepanel
+  // url.searchParams.append('templateType', 'dialog');
+  // const id = nanoid();
+  // searchParams.append('windowId', String(id));
+  // We use HashRouter currently
+  if (searchParams) {
```

Comment:
As I see there is only one case of this function call and searchParams are always null

Reply:
This is still a reasonable api I think